cmake/common/px4_base.cmake

```diff
@@ -507,7 +506,11 @@ function(px4_add_common_flags)
 		#               but generates too many false positives
 		)
 
-	if (NOT ${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
+	if (${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
+		list(APPEND warnings
+			-Wno-unused-const-variable
+		)
+	else(${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
```

Comment:
See cmake coding standard, because this is confusing, the typical procedure is to never do that:

```cmake
if (condition)
elseif(condition)
else()
endif()
```
Is the new standard way.